src/canvas.class.js

```diff
@@ -64,7 +64,56 @@
       this.requestRenderAllBound = this.requestRenderAll.bind(this);
       this._initStatic(el, options);
       this._initInteractive();
-      this._createCacheCanvas();
+      this._createCacheCanvas(); 
+
+      this._initHighlightDrawing();
+    },
+
+    // In-memory canvas element to support highlight drawing
+    highlightDrawingCanvasEl: null,
+
+    _initHighlightDrawing: function() {
+      console.log('fabric.Canvas::_initHighlightDrawing');
+      
+      //(1) Init off-screen buffer
+      if (!this.highlightDrawingCanvasEl) {
+        var canvasBuffer = document.createElement('canvas');
+        canvasBuffer.width  = this.width;
+        canvasBuffer.height = this.height;
+        this.highlightDrawingCanvasEl = canvasBuffer;
+      }
+
+      //(2) Insert Image representing buffer onto the stage
+      if (this.getHighlightLayerIndex() < 0) {
+        var fabricImageBuffer = new fabric.Image(this.highlightDrawingCanvasEl);
+        fabricImageBuffer.selectable = false;
+        fabricImageBuffer.isHighlightLayer = true;
```

Comment:
There's a way to indicate which object properties are serialized, if we want to either ignore this or treat it specially during deserializing we will need to add .isHighlightLayer to those serialized properties.

Reply:
Yeah, good eye ;)

My current thinking for serialization/deserialization would be to:

1. Allow images to get serialized again
2. If isHightlightLayer is set, have the image not set a src field
3. On deserialization, if isHighlightLayer is set, trigger this method. I tried to write it in a way where it should be fine to call again and again if needed.

So yeah, in this setup we'd need to add this property to the serialized list like you say.